### DISCUSSION TITLE
how to map tab to use vim-matchup

### DISCUSSION THREAD

**Discussion by WingDust** [andymass/vim-matchup#254] [Q&A]
how to map tab to use vim-matchup
in neovim ,I already use lua to map `tab` to `%`, but not all featurn can be use, How to do that?
```lua

local function map(mode, lhs, rhs, opts)
  local options = { noremap = true, silent = true }
  if opts then
    options = vim.tbl_extend('force', options, opts)
  end
  vim.api.nvim_set_keymap(mode, lhs, rhs, options)
end

map("n","<tab>","%",opt)

```

**Comment by andymass** [comment]
You should map to `<Plug>(matchup-%)` as follows:

    map("n", "<tab>" , "<Plug>(matchup-%)", { noremap = false })

Note you must use noremap = false.

**Reply by WingDust** [reply]
@andymass how to make `vi<tab>` also work like `vi%`
